app/mutations/addresses/release.rb

```diff
@@ -31,10 +32,18 @@ def execute
         info "Skip gateway address=#{address.id} in pool=#{@pool.id}"
 
       else
-        info "Delete address=#{address.id} in pool=#{@pool.id}"
-
-        address.delete!
+        if ping?(address.address)
+          info "#{self.address} still responds to ping, deleting omitted"
```

Comment:
And this should also add an error, so that the `Addresses::Release` fails. That way this will also show up in the agent logs, where it's easier to find.